doc/rtd/topics/code_review.rst

```diff
@@ -0,0 +1,170 @@
+   that, in GitHub terms, this is setting an Assignee, not requesting a
+   review.
+
+2. That **Committer** reviews the PR, resulting in one of the
+   following:
+
+   Approve
+     The **Committer** will squash merge immediately.
+
+     There may be circumstances where a PR should not be merged
+     immediately; these will be handled case-by-case, and the
+     **Proposer** should make this requirement very clear in their pull
+     request description.
+
+     The review process is **DONE**.
+
+   Approve (with nits)
+     If the **Proposer** submits their PR with "Allow edits from
+     maintainer" enabled, and the only changes the **Committer**
+     requests are minor "nits", the **Committer** can push fixes for
+     those nits and squash merge.  (If a **Proposer** wants to opt-out
+     of this, then they should uncheck "Allow edits from maintainer"
+     when submitting their PR.)
+
+     A nit is understood to be something like a minor style issue or a
+     spelling error, generally confined to a single line of code.
+
+     If a **Committer** is unsure as to whether their requested change
+     is a nit, they should not treat it as a nit.
+
+     The review process is **DONE**.
+
+   Outright rejection
+     The **Committer** will close the PR, with useful messaging for the
+     **Proposer** as to why this has happened.
+
+     This is reserved for cases where the proposed change is completely
+     unfit for landing, and there is no reasonable path forward.  This
+     should only be used sparingly, as there are very few cases where
+     proposals are completely unfit.
+
+     If a different approach to the same problem is planned, it should
+     be submitted as a separate PR.  The **Committer** should include
+     this information in their message when the PR is closed.
+
+     The review process is **DONE**.
+
+   Needs Changes
+     The **Committer** will give the **Proposer** a clear idea of what
+     is required for an Approve vote or, for more complex PRs, what the
+     next steps towards an Approve vote are.
+
+     The **Proposer** will ask questions if they don't understand, or
+     disagree with, the **Committer**'s review comments.
+
+     Once consensus has been reached, the **Proposer** will address the
+     review comments.
+
+     Once the review comments are addressed (as well as, potentially,
+     in the interim), CI will run.  If CI fails, the **Proposer** is
+     expected to fix CI failures.  If CI passes, the **Proposer**
+     should indicate that the PR is ready for re-review (by @ing the
+     assigned reviewer), effectively moving back to the start of this
+     section.
+
+Inactive Pull Requests
+======================
+
+PRs may be closed if they have been waiting on **Proposer** action for
+a certain amount of time without activity.  A PR will be marked as
+stale (with an explanatory comment) after 14 days of inactivity.  It
+will be closed after a further 7 days of inactivity.
```

Comment:
Do we have a way to avoid closing when we are waiting on the reviewer?

e.g. My PR needed fixes, I fixed them and @you to get your attention, but you didn't take a look after 14 days.

Reply:
I don't think we have a robust way of doing it presently, no. A later iteration of this, with automation, might have labels that we could use to disable the messaging.

Perhaps we should include something like "(If this PR has been waiting on the _reviewer_ for 14 days, please ping @powersj who will apologise and ensure someone looks at it.)" in the automated messaging, as a backstop?

Reply:
:+1: I really like that idea of escalating to me

Reply:
I've added this to the relevant PR: https://github.com/canonical/cloud-init/pull/125/files#diff-12eea56dbf5e1eb25defd0d34cc314bbR21